templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -958,6 +977,88 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose &ik_pose,
   return false;
 }
 
+// Used when there are no redundant joints - aka no free params
+bool IKFastKinematicsPlugin::getPositionIK(const std::vector<geometry_msgs::Pose> &ik_poses,
+                                           std::vector< std::vector<double> >& solutions,
+                                           kinematics::KinematicsResult& result,
+                                           const kinematics::KinematicsQueryOptions &options) const
+{
+  ROS_DEBUG_STREAM_NAMED("ikfast","getPositionIK with multiple solutions");
+
+  if(!active_)
+  {
+    ROS_ERROR("kinematics not active");
+    return false;
+  }
+
+  std::vector<double> vfree(free_params_.size());
+  const std::vector<double> ik_seed_state(num_joints_);
+  for(std::size_t i = 0; i < free_params_.size(); ++i)
+  {
+    int p = free_params_[i];
+    ROS_ERROR("%u is %f",p,ik_seed_state[p]);  // DTC
+    vfree[i] = ik_seed_state[p];
```

Comment:
In the original getPositionIK(), the vfree array is populated by indexing into the ik_seed_state that is passed as an argument.  In this case, it will default to zero so the solutions will have the free joint fixed at that value.  
Perhaps using the joint discretization members to populate vfree would be more appropriate.